segbot_sensors/src/segbot_sensors/battery_monitor.cpp

```diff
@@ -0,0 +1,127 @@
+#include "ros/ros.h"
+#include "ros/time.h"
+#include <vector>
+#include "std_msgs/Header.h"
+#include "std_msgs/Float32.h"
+#include "diagnostic_msgs/DiagnosticStatus.h"
+#include "diagnostic_msgs/DiagnosticArray.h"
+#include "diagnostic_msgs/KeyValue.h"
+#include "smart_battery_msgs/SmartBatteryStatus.h" //file found
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+float voltage;
+bool sentMail = false;
+void voltageCallback(const smart_battery_msgs::SmartBatteryStatus::ConstPtr& msg)
+{
+  voltage = (msg->voltage);
+ // ROS_INFO("I heard: [%f]", msg->voltage);
+  //todo: add voltage profiler logic - write profile rates in /config
+}
+
+int sendMail()
+{
+    char msg[] = "Robot battery low. Please seek robot and bring to lab for charging.\nNOTE:This is an automated message generated by the diagnostic aggregator node. Do NOT reply to this message.";
+    int retval = -1;
+    FILE *mailpipe = popen("/usr/lib/sendmail -t", "w");
+    if (mailpipe != NULL) {
+        fprintf(mailpipe, "To: maxsvetlik@gmail.com\n");
+        fprintf(mailpipe, "From: BWIrobotAlert@bwi.utexas.edu\n");
+        fprintf(mailpipe, "Subject: Segbot Low Battery Alert\n");
+        fwrite(msg, 1, strlen(msg), mailpipe);
+        fwrite(".\n", 1, 2, mailpipe);
+        pclose(mailpipe);
+        retval = 0;
+     }
+     else {
+         perror("Failed to invoke sendmail");
+     }
+     return retval;
+}
+
+
+int main(int argc, char **argv)
+{
+  ros::init(argc, argv, "battery_estimator");
```

Comment:
You've got uneven indentation here. 2 spaces here vs 4 spaces above. Could you use 2 throughout?